src/core/src/model.cpp

```diff
@@ -238,7 +247,33 @@ void ov::Model::validate_nodes_and_infer_types() const {
         } else if (const auto& read_value = std::dynamic_pointer_cast<ngraph::op::ReadValueBase>(node)) {
             pair_checker[read_value->get_variable().get()].cnt_read_val++;
         }
+
+        // Detect names collisions only for nodes with autogenerated names
+        const auto friendly_name = node->get_friendly_name();
+        if (unique_friendly_names.find(friendly_name) == unique_friendly_names.end()) {
+            unique_friendly_names.insert(friendly_name);
+            if (node->m_friendly_name.empty())
```

Comment:
In some transformations we can accidentally set node friendly name to its name, so it will looks like user-defined name. I think we also need to consider name as autogenerated if `friendly_name == get_name()`

Reply:
Discussed offline.